ui/app/components/secure-variable-form/namespace-filter.hbs

```diff
@@ -0,0 +1,24 @@
+<Trigger @do={{this.fetchNamespaces}} @onSuccess={{this.formatAndSetNamespaces}} as |trigger|>
```

Comment:
Q: I'm not generally familiar with our `Trigger` component, but is it comparable to a `did-insert` render modifier?

Reply:
Not quite. The `Trigger` component is how we handle asynchronous behavior in a component. The `Trigger` component has a state machine, where we trigger the `onTrigger` callback function to run when the component mounts, and then we also allow subsequent callback functions to run based to the result of the Promise and that's why the component API has `onSuccess` and `onError` and it resets its state back to `idle` using `triggerTask` so we don't get stuck in a failed state.

We use `did-insert` to trigger a callback function to run when a component mounts. And I think in the future, we might be able to a template helper to trigger a fetch request directly in the template.